examples/vitals/pull_vitals.py

```diff
@@ -53,13 +53,80 @@
+    device = pw.devices[x].device.device
+    parent = str(device.componentParentDin.value)
+    vitals = pw.devices[x].vitals
+    alerts = pw.devices[x].alerts
+
+    name = str(device.din.value)
+    print("Device %d: %s " % (x, name))
+
+    # e.STSTSM = "STSTSM",
+    # e.POD = "TEPOD",
+    # e.PINV = "TEPINV",
+    # e.PVAC = "PVAC",
+    # e.PVS = "PVS",
+    # e.SYNC = "TESYNC",
+    # e.MSA = "TEMSA",
+    # e.NEURIO = "NEURIO",
+    # e.ACPW = "ACPW",
+    # e.PVI = "PVI",
+    # e.SPW = "SPW"
+
+    if name.startswith("TETHC--"):
+        print("   - Inverter")
+
+    if device.HasField("partNumber"):
+        print("   - Part Number: %s" % device.partNumber.value)
+    if device.HasField("serialNumber"):
+        print("   - Serial Number: %s" % device.serialNumber.value)
+    if device.HasField("manufacturer"):
+        print("   - Manufacturer: %s" % device.manufacturer.value)
+    if device.HasField("siteLabel"):
+        print("   - Site Label: %s" % device.siteLabel.value)
+    if device.HasField("componentParentDin"):
+        print("   - Parent DIN: %s" % device.componentParentDin.value)
+    if device.HasField("firmwareVersion"):
+        print("   - Firmware Version: %s" % device.firmwareVersion.value)
+    if device.HasField("firstCommunicationTime"):
+        print("   - First Communicated At: %s" % device.firstCommunicationTime.ToDatetime())
+    if device.HasField("lastCommunicationTime"):
+        print("   - Last Communicated At: %s" % device.lastCommunicationTime.ToDatetime())
+    # if device.HasField("connectionParameters"):
+    #    print("   - Connection Parameters: %s" % device.connectionParameters)
+
+    if device.HasField("deviceAttributes"):
+        attributes = device.deviceAttributes
+        if attributes.HasField("teslaEnergyEcuAttributes"):
+            print("   - Ecu:")
+            print("       - type: %i" % attributes.teslaEnergyEcuAttributes.ecuType)
+        if attributes.HasField("generatorAttributes"):
+            print("   - Generator:")
+            print("       - nameplateRealPowerW: %i" % attributes.generatorAttributes.nameplateRealPowerW)
+            print("       - nameplateApparentPowerVa: %i" % attributes.generatorAttributes.nameplateApparentPowerVa)
+        if attributes.HasField("pvInverterAttributes"):
+            print("   - Inverter:")
+            print("       - nameplateRealPowerW: %i" % attributes.pvInverterAttributes.nameplateRealPowerW)
+        if attributes.HasField("meterAttributes"):
+            print("   - Meter:")
+            for location in attributes.meterAttributes.meterLocation:
+                print("       - location: %i" % location)
+    a = 0
+
+    while (a < len(alerts)):
+        if (a == 0):
+            print("   - Alerts:")
+
+        print("       - ALERT_%i = %s" % (a, alerts[a]) )
+        a += 1
+
+    print("   - Vitals:")
+
     for y in pw.devices[x].vitals:
         vital_name = str(y.name)
+        if (y.HasField('intValue')):
```

Comment:
I don't actually receive any that have this, but it is in the spec, so it might as well be handled.

Reply:
Same here, but I agree.  Would love to hear from others if you are getting these data points.